### PR TITLE
feat(minReadySeconds): support minReadySeconds in TC annotation

### PR DESCRIPTION
**Pull request by liubog2008**
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

File: pkg/manager/member/tidb_upgrader.go [hunk 1]
```diff
@@ -100,7 +119,7 @@ func (u *tidbUpgrader) Upgrade(tc *v1alpha1.TidbCluster, oldSet *apps.StatefulSe
 		}
 
 		if revision == tc.Status.TiDB.StatefulSet.UpdateRevision {
-			if !podutil.IsPodReady(pod) {
+			if !podutil.IsPodAvailable(pod, int32(minReadySeconds), metav1.Now()) {
```
Comment (csuzhangxc) [inline pkg/manager/member/tidb_upgrader.go:122]: do we need to update the error in L123?

File: pkg/manager/member/tidb_upgrader.go [hunk 2]
```diff
@@ -15,17 +15,25 @@ package member
...
 	apps "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog/v2"
 	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
 )
 
+const (
+	// TODO: change to use minReadySeconds in sts spec
+	// See https://kubernetes.io/blog/2021/08/27/minreadyseconds-statefulsets/
+	annoKeyTiDBMinReadySeconds = "tidb.pingcap.com/tidb-min-ready-seconds"
```
Comment (KanShiori) [inline pkg/manager/member/tidb_upgrader.go:34]: Do we need to put it in `apis` pkg and expose this var?

Comment (liubog2008) [reply pkg/manager/member/tidb_upgrader.go:34]: I hope to use it only in inner pkg and export it after https://kubernetes.io/blog/2021/08/27/minreadyseconds-statefulsets/ is ready for us.

File: pkg/manager/member/tidb_upgrader.go (hunk 1 above)
Comment (liubog2008) [reply pkg/manager/member/tidb_upgrader.go:122]: fixed

**Review by KanShiori — APPROVED** [review]

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**